jmeter-service/jmeterUtils.go

```diff
@@ -9,117 +9,59 @@ import (
+func DoRemoveTempFiles() bool {
+	debugFlag := os.Getenv("DEBUG_KEEP_TEMP_FILES")
+	if strings.Compare(debugFlag, "true") == 0 {
+		return false
 	}
-	return "localhost:8080"
-}
 
-func executeJMeter(testInfo *TestInfo, workload *Workload, resultsDir string, url *url.URL, LTN string, funcValidation bool, logger *keptnutils.Logger) (bool, error) {
-	os.RemoveAll(resultsDir)
-	os.MkdirAll(resultsDir, 0644)
-
-	resourceHandler := configutils.NewResourceHandler(getConfigurationServiceURL())
-
-	// =====================================
-	// implementing - https://github.com/keptn-contrib/jmeter-extended-service/issues/3
-	// trying to load script from service, then stage and last from project
-	// if test script cannot be found we skip execution
-	testScriptResource, err := resourceHandler.GetServiceResource(testInfo.Project, testInfo.Stage, testInfo.Service, workload.Script)
-	if err != nil && err == configutils.ResourceNotFoundError {
-		// if not found on serivce level - lets try it on stage level
-		testScriptResource, err = resourceHandler.GetStageResource(testInfo.Project, testInfo.Stage, workload.Script)
-
-		if err != nil && err == configutils.ResourceNotFoundError {
-			// if not found on stage level we try project level
-			testScriptResource, err = resourceHandler.GetProjectResource(testInfo.Project, workload.Script)
-
-			if err != nil && err == configutils.ResourceNotFoundError {
-				logger.Debug("Skipping test execution because " + workload.Script + " not found on service, stage or project level.")
-				return true, nil
-			} else if err != nil {
-				logger.Error("Could not fetch testing script: " + err.Error())
-				return false, err
-			}
-		} else if err != nil {
-			logger.Error("Could not fetch testing script: " + err.Error())
-			return false, err
-		}
-	} else if err != nil {
-		logger.Error("Could not fetch testing script: " + err.Error())
-		return false, err
-	}
+	return true
+}
 
-	os.RemoveAll(workload.Script)
-	pathArr := strings.Split(workload.Script, "/")
-	directory := ""
-	for _, pathItem := range pathArr[0 : len(pathArr)-1] {
-		directory += pathItem + "/"
-	}
+// ToString returns a string representation of a TestInfo object
+func (ti *TestInfo) ToString() string {
+	return fmt.Sprintf("Project: %s, Service: %s, Stage: %s, TestStrategy: %s, Context: %s", ti.Project, ti.Service, ti.Stage, ti.TestStrategy, ti.Context)
+}
 
-	err = os.MkdirAll(directory, os.ModePerm)
-	if err != nil {
-		return false, err
-	}
-	testScriptFile, err := os.Create(workload.Script)
-	if err != nil {
-		logger.Error(err.Error())
-		return false, err
-	}
-	defer testScriptFile.Close()
+/**
+ * Returns additoinal JMeter Command Line Parameters including additional params passed to the JMeter script
+ */
+func addJMeterCommandLineArguments(testInfo *TestInfo, initialList []string) []string {
+	dtTenant := fmt.Sprintf("-JDT_TENANT=%s", os.Getenv("DT_TENANT"))
+	dtAPIToken := fmt.Sprintf("-JDT_API_TOKEN=%s", os.Getenv("DT_API_TOKEN"))
```

Comment:
Not sure if this is ideal. In dynatrace-service we use `dynatrace.conf` to determine the secret that should be used. Here we refer to an environment variable based on a static secret.

Reply:
yes - because I didnt want to have a dependency to dynatrace.conf.yaml but simply allow additonal "secrets" to be passed to the JMEter script. I think this should to included into our general discussion of passing "secrets/endpoint configuraitons" to any type of keptn service